lib/common.py

```diff
@@ -119,41 +119,40 @@ def timeout(proc):
   #
   # When we use "exec" to run the "command". This will cause the "command" to
   # inherit the shell process and proc.pid will represent the actual "command".
-  proc = subprocess.Popen(
-    "exec " + command,
-    stdin = subprocess.PIPE,
-    stdout = subprocess.PIPE,
-    stderr = subprocess.STDOUT,
-    shell = True)
-
-  logger("CALL: {} (pid: {})".format(command, proc.pid))
-
-  reader = threading.Thread(target = readproc, args = [proc])
-  timer = threading.Timer(get_media()._get_call_timeout(), timeout, [proc])
-  reader.daemon = True
-  timer.daemon = True
-  reader.start()
-  timer.start()
-
-  try: # in case of user interrupt
-    proc.wait()
-    timer.cancel()
-  except:
-    killproc(proc)
-    raise
-  finally:
-    timer.cancel()
-    timer.join(30)
-    reader.join(30)
-
-  if timeout.triggered:
-    error = True
-    get_media()._report_call_timeout()
-    message = "CALL TIMEOUT: timeout after {} seconds (pid: {}).".format(
-      timer.interval, proc.pid)
-  elif proc.returncode != 0:
-    error = True
-    message = "CALL ERROR: failed with exitcode {} (pid: {})".format(proc.returncode, proc.pid)
+  with subprocess.Popen("exec " + command,
```

Comment:
This is not backwards compatible with python 2.  This means our internal smoke/full test configs will need to be updated to python3 and merged atomically with this PR.

I'd prefer we keep all python3 changes backwards compatible with python2 for at least a few months... so any external dependents (i.e. smoke/full or other users?) have time to migrate.

So for now, it would be better to keep the original syntax (i.e. don't use `with` on popen).  Instead, to satisfy both py2 and py3, all we need to do is call `proc.stdin.close()` and `proc.stdout.close()` in the `finally:` block after `reader.join(30)`.

Reply:
will try on local first